src/amo/components/InstallWarning/index.js

```diff
@@ -0,0 +1,184 @@
+/* @flow */
+import * as React from 'react';
+import { connect } from 'react-redux';
+import { compose } from 'redux';
+
+import type { AppState } from 'amo/store';
+import { ADDON_TYPE_EXTENSION, UNINSTALLED, UNKNOWN } from 'core/constants';
+import translate from 'core/i18n/translate';
+import log from 'core/logger';
+import tracking from 'core/tracking';
+import { withExperiment } from 'core/withExperiment';
+import Notice from 'ui/components/Notice';
+import type { AddonType } from 'core/types/addons';
+import type { I18nType } from 'core/types/i18n';
+import type { InstalledAddon } from 'core/reducers/installations';
+import type { WithExperimentInjectedProps } from 'core/withExperiment';
+
+import './styles.scss';
+
+type Props = {|
+  addon: AddonType,
+|};
+
+type CouldShowWarningParams = {|
+  addonIsExtension: boolean,
+  addonIsRecommended: boolean | void,
+  experimentEnabled: $PropertyType<
+    WithExperimentInjectedProps,
+    'experimentEnabled',
+  >,
+  installStatus: $PropertyType<InstalledAddon, 'status'>,
+|};
+
+export const couldShowWarning = ({
+  addonIsExtension,
+  addonIsRecommended,
+  experimentEnabled,
+  installStatus,
+}: CouldShowWarningParams) => {
+  return (
+    addonIsExtension &&
+    !addonIsRecommended &&
+    experimentEnabled &&
+    [UNINSTALLED, UNKNOWN].includes(installStatus)
+  );
+};
+
+const addonIsExtension = (addon: AddonType) => {
+  return addon.type === ADDON_TYPE_EXTENSION;
+};
+
+type InternalProps = {|
+  ...Props,
+  ...WithExperimentInjectedProps,
+  _couldShowWarning: typeof couldShowWarning,
+  _log: typeof log,
+  _tracking: typeof tracking,
+  i18n: I18nType,
+  installStatus: $PropertyType<InstalledAddon, 'status'>,
+|};
+
+export const EXPERIMENT_CATEGORY_CLICK =
+  'AMO Install Button Warning Experiment - Click';
+export const EXPERIMENT_CATEGORY_SHOW =
```

Comment:
I suggest we use the same verb in both the constant and its value, so if it's `show` in the constant name, it should be `Show` in the value and not `Display`.